crates/data_components/src/graphql/provider.rs

```diff
@@ -88,6 +92,21 @@ impl GraphQLTableProviderBuilder {
             )
             .await?;
 
+        if result.records.is_empty() {
```

Comment:
If we have a health check query, we should just use that directly instead of trying the other query first.